content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -1,16 +1,168 @@
 
 </TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create custom queries
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an [extended client](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions). An extended client is a variant of the standard Prisma Client that is wrapped by one or more extensions.
+
+Use the `query` extension component to create custom queries. You can add a custom query to the following:
+
+- [A specific operation in a specific model of your schema](#add-a-custom-query-to-a-specific-operation-in-a-specific-model)
+- [A specific operation in all models of your schema](#add-a-custom-query-to-a-specific-operation-in-all-models-of-your-schema)
+- [All operations in a specific model of your schema](#add-a-custom-query-to-all-operations-in-a-specific-model)
+- [All operations in all models of your schema](#add-a-custom-query-to-all-operations-in-all-models-of-your-schema)
+
+To create a custom query, use the following structure:
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  query?: {
+    user: { ... } //in this case, we add a query to the `user` model
+  },
+});
+```
+
+The properties are as follows:
+
+- `name`: (optional) specifies a name for the extension that appears in error logs.
+- `query`: (optional) defines a custom query.
+
+### Add a custom query to a specific operation in a specific model
+
+The `query` object contains functions that map to the names of the [Prisma Client operations](/docs/reference/api-reference/prisma-client-reference#model-queries), such as `findUnique`, `findFirst`, `findMany`, `count`, and `create`. The following example defines a custom query that finds users who are older than 18 years:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findMany({ model, operation, args, data }) {
+        // set `age` and fill with the rest of `where`
+        args.where = { age: { gt: 18 }, ...args.where }
+        const users = await data
+        return data
+      },
+    },
+  },
+})
+
+await xprisma.user.findMany() // returns users whose age is greater than 18
+```
+
+In the above example, a call to `xprisma.user.findMany` triggers `query.user.findMany`. Each callback receives a type-safe `{ model, operation, args, data }` object that describes the query. This object has the following properties:
```

Comment:
Does this really trigger `query.user.findMany`? (should it say `prisma.user.findMany`?)
What is the callback here?

Reply:
@millsp  Could you comment on this, please?

Reply:
It lgtm. You call `xprisma.user.findMany` then it triggers the query extension at `query.user.findMany`

Reply:
Thanks!